projects/saturn/src/main/java/io/fairspace/saturn/services/views/SparqlQueryService.java

```diff
@@ -169,14 +195,10 @@ private ValueDTO toValueDTO(RDFNode node) {
         return new ValueDTO(label, resource.getURI());
     }
 
-    private Query getQuery(CountRequest request) {
+    private Query getQuery(CountRequest request, boolean isCount) {
         var view = getView(request.getView());
 
-        var builder = new StringBuilder("PREFIX fs: <")
-                .append(FS.NS)
-                .append(">\n\nSELECT ?")
-                .append(view.name)
-                .append("\nWHERE {\n");
+        var builder = new StringBuilder().append("SELECT ").append("%s").append("\nWHERE {\n");
```

Comment:
why this change? Why not putting the view name already here, instead of the '%s'? And why adding the prefix later?

Reply:
ok, I understand now, but the query building logic is now hard to follow...

Reply:
agree, slightly improved, but the prefix to be added in the end anywhere due to potential query wrapping for count query. What do you think now?

Reply:
sure, looks good!